### PR TITLE
PowerVS additional events

### PR DESCRIPTION
**Pull request by jaywcarman**
Requires:
- https://github.com/ManageIQ/manageiq-content/pull/702

### CODE REVIEW

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb [hunk 1]
```diff
@@ -6,21 +6,48 @@ def self.event_to_hash(event, ems_id)
...
       parse_vm_event!(event, event_hash)
+    when /^volume/
+      parse_volume_event!(event, event_hash)
     end
 
     event_hash
   end
 
+  def self.parse_image_event!(event, event_hash)
+    if ['delete'].include?(event[:action])
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb:30]: I'm guessing there are going to be more actions to be added here in the future?

Comment (agrare) [reply app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb:30]: Or just staying consistent with the other `parse_*_event!` methods?

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_target_parser.rb [hunk 2]
```diff
@@ -9,6 +9,25 @@ def parse
...
+        :manager_ref => {:ems_ref => ems_event[:vm_ems_ref]},
+        :manager_id  => ems_event.ext_management_system.id,
+        :event_id    => ems_event.id
+      )
+    when /^image/
+      if ems_event[:message].include?("successfully imported into image catalog")
+        targets << ems_event.ext_management_system
+      end
+    when /^network/
+      network_manager = ManageIQ::Providers::IbmCloud::PowerVirtualServers::NetworkManager.find_by(:parent_ems_id => ems_event[:ems_id])
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_target_parser.rb:24]: `ems_event` should be an EmsEvent object, which `ext_management_system` is already a loaded association for, so you should be able to do `ems_event.ext_management_system.network_manager` without doing the query manually.

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_target_parser.rb [hunk 3]
```diff
@@ -9,6 +9,25 @@ def parse
     targets = []
 
     case ems_event[:event_type]
+    when /^image\.delete/
+      targets << InventoryRefresh::Target.new(
+        :association => :miq_templates,
+        :manager_ref => {:ems_ref => ems_event[:vm_ems_ref]},
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_target_parser.rb:15]: It is minior, but prefer to use the accessor methods like `ems_event.vm_ems_ref` so it is more obvious that this isn't just a hash

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_target_parser.rb [hunk 4]
```diff
@@ -20,6 +39,9 @@ def parse
         :manager_id  => ems_event.ext_management_system.id,
         :event_id    => ems_event.id
       )
+    when /^volume/
+      storage_manager = ManageIQ::Providers::IbmCloud::PowerVirtualServers::StorageManager.find_by(:parent_ems_id => ems_event[:ems_id])
+      targets << storage_manager
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_target_parser.rb:44]: Similar here, you can do `targets << ems_event.ext_management_system.storage_manager`

**Comment by agrare** [comment]
Love seeing this getting filled out more, just a few minor things

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_target_parser.rb (hunk 2 above)
Comment (jaywcarman) [reply app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_target_parser.rb:24]: Nice! I'm glad to get rid of the queries.

File: app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb (hunk 1 above)
Comment (jaywcarman) [reply app/models/manageiq/providers/ibm_cloud/power_virtual_servers/cloud_manager/event_parser.rb:30]: There are quite a few `image` events available today, but unfortunately only `delete` provides enough metadata to trigger a refresh (well, actually I'm parsing the message in the `EventTargetParser` to trigger a refresh when an image import is completed - [ref](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/335/files#diff-d14118ccd17b8c31d6b54be4d044f9d6d9902de875fadefe758a35cae6f62f01R20-R22)). I'm hoping in the future they'll standardize the event responses and we can do more here.

**Review by agrare — APPROVED** [review]
Looking good

**Pull request merged**